includes/payments/functions.php

```diff
@@ -1430,11 +1430,19 @@ function edd_set_payment_transaction_id( $payment_id = 0, $transaction_id = '' )
  */
 function edd_get_purchase_id_by_key( $key ) {
 	global $wpdb;
+	$global_key_string = 'edd_purchase_id_by_key' . $key;
+	global $$global_key_string;
```

Comment:
@pippinsplugins yeah, I'm not a huge fan of the variable variables...but I think it's just the simplest method here.